packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -693,6 +693,38 @@ describe('FocusScope', function () {
         await waitFor(() => expect(document.activeElement).toBe(focusable));
       });
     });
+
+    it('should not not restore focus when active element is outside the scope', function () {
+      function Test() {
+        const [display, setDisplay] = useState(false);
+
+        return (
+          <div>
+            <input data-testid="trigger" onFocus={() => setDisplay(state => !state)} />
+            {display &&
+              <FocusScope autoFocus restoreFocus>
```

Comment:
not quite sure this is testing the right thing

I feel like what we want is closer to this:
```
  <div>
      <button type="button" onClick={() => setDisplay(state => !state)}>
        {display ? 'Close dialog' : 'Open dialog'}
      </button>
      <button type="button" onClick={() => setDisplay(state => !state)}>
        {display ? 'Close dialog' : 'Open dialog'}
      </button>
      {display &&
        <FocusScope restoreFocus>
          <div role="dialog">
            <input  />
            <input  />
            <input  />
          </div>
        </FocusScope>
      }
    </div>
...
userEvent.click button1
act(jest run all timers)
expect document.activeElement to be button1
userEvent.click button2
act(jest run all timers)
expect document.activeElement to be button2
```

 The reason for two buttons is to verify that restore focus didn't kick in and restore it back to button1 on FocuseScope unmount